lib/rubocop/cop/rails/require_dependency.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for the usage of `require_dependency`.
+      # Applications running in zeitwerk mode should not use require_dependency
+      #
+      # @example
+      #   # bad
+      #   require_dependency 'some_lib'
+      class RequireDependency < Base
+        MSG = 'Applications running in zeitwerk mode should not use `require_dependency`.'
+
```

Comment:
Zweitwerk has been introduced in Rails 6.0. Can you add `minimum_target_rails_version 6.0`?
https://github.com/rubocop/rubocop-rails/blob/v2.9.1/lib/rubocop/cop/rails/pick.rb#L22-L27

And Can you add some tests as well?
https://github.com/rubocop/rubocop-rails/blob/v2.9.1/spec/rubocop/cop/rails/pick_spec.rb#L6